app/gui/view/graph-editor/src/builtin/visualization/java_script/table.js

```diff
@@ -95,341 +49,197 @@ class TableVisualization extends Visualization {
-            color: rgba(0, 0, 0, 0.7);
-            padding: 0;
-        }
-
-        td.plaintext,
-            th {
-            padding: 5px;
-        }
-
-        th,
-            td {
-            border: 1px solid transparent;
-            background-clip: padding-box;
-        }
-
-        th, .hiddenrows {
-            color: rgba(0, 0, 0, 0.9);
-            font-weight: 400;
-        }
-
-        td {
-            background-color: rgba(0, 0, 0, 0.025);
-        }
-
-        th {
-            background-color: rgba(30, 30, 20, 0.1);
-        }
+        if (!this.tabElem) {
+            while (this.dom.firstChild) {
+                this.dom.removeChild(this.dom.lastChild)
+            }
 
-        .hiddenrows {
-            margin-left: 5px;
-            margin-top: 5px;
+            const style =
+                '.ag-theme-alpine { --ag-grid-size: 3px; --ag-list-item-height: 20px; display: inline; }'
+            const styleElem = document.createElement('style')
+            styleElem.innerHTML = style
+            this.dom.appendChild(styleElem)
+
+            const tabElem = document.createElement('div')
+            tabElem.setAttributeNS(null, 'id', 'vis-tbl-view')
+            tabElem.setAttributeNS(null, 'class', 'scrollable ag-theme-alpine')
+            this.dom.appendChild(tabElem)
+            this.tabElem = tabElem
+            this.updateTableSize()
+
+            this.agGridOptions = {
+                rowData: [],
+                columnDefs: [],
+                defaultColDef: {
+                    editable: false,
+                    sortable: true,
+                    filter: true,
+                    resizable: true,
+                    minWidth: 50,
+                },
+                enableRangeSelection: true,
+            }
+            this.agGrid = new agGrid.Grid(tabElem, this.agGridOptions)
         }
-        </style>`
 
-        const tabElem = document.createElement('div')
-        tabElem.setAttributeNS(null, 'id', 'vis-tbl-view')
-        tabElem.setAttributeNS(null, 'class', 'scrollable')
-        tabElem.setAttributeNS(null, 'width', '100%')
-        tabElem.setAttributeNS(null, 'height', '100%')
-        this.tabElem = tabElem
-        this.dom.appendChild(tabElem)
-        this.updateTableSize()
+        let parsedData = typeof data === 'string' ? JSON.parse(data) : data
 
-        let parsedData = data
-        if (typeof data === 'string') {
-            parsedData = JSON.parse(data)
-        }
-
-        let style = style_light
-        if (document.getElementById('root').classList.contains('dark-theme')) {
-            style = style_dark
-        }
+        let columnDefs = []
+        let rowData = []
+        let dataTruncated = false
 
         if (parsedData.error !== undefined) {
-            tabElem.innerHTML = 'Error: ' + parsedData.error
+            this.agGridOptions.api.setColumnDefs([
+                {
+                    field: 'error',
+                    width: 1000,
```

Comment:
what is this number? Could you please refacto it to well named constant in the top of the file, please?